spk/ffsync/Makefile

```diff
@@ -1,47 +1,58 @@
 DISPLAY_NAME = Firefox Sync Server 1.5
-CHANGELOG = "1. Fix Python requirement<br>2. Fix startup issue when pidfile is not yet written"
+CHANGELOG = "1. Upgrade to Mozilla Sync Storage built with Rust."
 
-HOMEPAGE     = https://docs.services.mozilla.com/howtos/run-sync-1.5.html
-LICENSE      = MPL/GPL/LGPL
-HELPURL      = https://github.com/SynoCommunity/spksrc/wiki/Firefox-Sync-Server-1.5
+HOMEPAGE = https://mozilla-services.readthedocs.io/en/latest/howtos/run-sync-1.5.html
+LICENSE = MPL 2.0
 
-INSTALLER_SCRIPT = src/installer.sh
-SSS_SCRIPT       = src/dsm-control.sh
-FWPORTS          = src/${SPK_NAME}.sc
-CONF_DIR         = src/conf/
+WIZARDS_DIR = src/wizard/
 
-INSTALL_DEP_SERVICES = mysql
-START_DEP_SERVICES = mysql
+STARTABLE = yes
+SERVICE_USER = auto
+SERVICE_SETUP = src/service-setup.sh
+SERVICE_PORT = 8132
+SERVICE_PORT_TITLE = $(DISPLAY_NAME)
 
-INSTALL_PREFIX = /usr/local/$(SPK_NAME)
+# Admin link for in DSM UI
+ADMIN_PORT = $(SERVICE_PORT)
 
-POST_STRIP_TARGET = ffsync_extra_install
+# [cryptography]
+# We pin cryptography to 3.4.8 due to a build error that occurs with the
+# current version (36.0.0 at the time of this commit). In short, building
+# the cryptography package with Rust results in a segfault, so we need to
+# set the CRYPTOGRAPHY_DONT_BUILD_RUST env var to force the package to build
+# with C instead. This env var is only present in cryptography<3.5.
+ENV += CRYPTOGRAPHY_DONT_BUILD_RUST=1
 
-BUSYBOX_CONFIG  = usermng
-ENV += BUSYBOX_CONFIG="$(BUSYBOX_CONFIG)"
+# [mysqlclient]
+ENV += MYSQLCLIENT_CFLAGS="$(CFLAGS) -I$(STAGING_INSTALL_PREFIX)/include/mysql -I$(STAGING_INSTALL_PREFIX)/$(PYTHON_INC_DIR)"
+ENV += MYSQLCLIENT_LDFLAGS="$(LDFLAGS)"
+
+POST_STRIP_TARGET = ffsync_extra_install
 
 include ../../mk/spksrc.spk.mk
 
 .PHONY: ffsync_extra_install
 ffsync_extra_install:
-	install -m 755 -d $(STAGING_DIR)/var
-	install -m 644 src/ffsync.ini $(STAGING_DIR)/var/
-	sed -i -e "s|https://github.com/mozilla-services/mozservices/archive/e00e1b68130423ad98d0f6185655bde650443da8.zip|mozsvc==0.8|g" \
-	       -e "s|https://github.com/mozilla-services/tokenserver/archive/d7e513e8a4f5c588b70d685a8df1d2e508c341c0.zip|tokenserver==1.2.7|g" \
-	       -e "s|https://github.com/mozilla-services/server-syncstorage/archive/1.5.5.zip|SyncStorage==1.5.5|g" \
-	       -e "s|https://github.com/mozilla-services/syncserver/archive/1.5.2.tar.gz|syncserver==1.5.2|g" \
-                $(STAGING_DIR)/share/wheelhouse/requirements.txt
+	@$(MSG) Install config file
+	@install -m 755 -d $(STAGING_DIR)/var
+	@install -m 644 src/var/local.toml $(STAGING_DIR)/var/
+	@$(MSG) Patch binaries built with rust to find shared libraries
+	@patchelf --set-rpath /var/packages/$(SPK_NAME)/target/lib:/var/packages/$(PYTHON_PACKAGE)/target/lib $(STAGING_DIR)/bin/syncserver
```

Comment:
@mreid-tt you have to devine PYTHON_PACKAGE, otherwise the syncserver will not find libpython* that is located the installed python311 package

But why did you remove the use of prebuilt python?

Reply:
I removed it because of this: https://github.com/SynoCommunity/spksrc/pull/5942#discussion_r1412823911